### ISSUE TITLE
Argument 3 passed to Sonata\Twig\Extension\TemplateExtension::__construct() must be an instance of Sonata\CoreBundle\Model\Adapter\AdapterInterface, instance of Sonata\Doctrine\Adapter\AdapterChain given

### ISSUE THREAD

**Issue by axzx** [sonata-project/SonataCoreBundle#613]
#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.42.2 3.42.2 The missing Symfony Admin Generator
sonata-project/block-bundle              3.13.0 3.13.0 Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  2.0.1  Cache library
sonata-project/core-bundle               3.13.2 3.13.2 Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.4.0  2.4.0  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.1.2  1.1.2  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.6.3  3.6.3  Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/exporter                  1.10.0 1.10.0 Lightweight Exporter library
```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
symfony/apache-pack           v1.0.1  v1.0.1  A pack for Apache support in Symfony
symfony/asset                 v4.2.1  v4.2.1  Symfony Asset Component
symfony/browser-kit           v4.2.1  v4.2.1  Symfony BrowserKit Component
symfony/cache                 v4.2.1  v4.2.1  Symfony Cache component with PSR-6, PSR-16, and tags
symfony/config                v4.2.1  v4.2.1  Symfony Config Component
symfony/console               v4.2.1  v4.2.1  Symfony Console Component
symfony/contracts             v1.0.2  v1.0.2  A set of abstractions extracted out of the Symfony components
symfony/css-selector          v4.2.1  v4.2.1  Symfony CssSelector Component
symfony/debug                 v4.2.1  v4.2.1  Symfony Debug Component
symfony/debug-bundle          v4.2.1  v4.2.1  Symfony DebugBundle
symfony/debug-pack            v1.0.7  v1.0.7  A debug pack for Symfony projects
symfony/dependency-injection  v4.2.1  v4.2.1  Symfony DependencyInjection Component
symfony/doctrine-bridge       v4.2.1  v4.2.1  Symfony Doctrine Bridge
symfony/dom-crawler           v4.2.1  v4.2.1  Symfony DomCrawler Component
symfony/dotenv                v4.2.1  v4.2.1  Registers environment variables from a .env file
symfony/event-dispatcher      v4.2.1  v4.2.1  Symfony EventDispatcher Component
symfony/expression-language   v4.2.1  v4.2.1  Symfony ExpressionLanguage Component
symfony/filesystem            v4.2.1  v4.2.1  Symfony Filesystem Component
symfony/finder                v4.2.1  v4.2.1  Symfony Finder Component
symfony/flex                  v1.1.8  v1.1.8  Composer plugin for Symfony
symfony/form                  v4.2.1  v4.2.1  Symfony Form Component
symfony/framework-bundle      v4.2.1  v4.2.1  Symfony FrameworkBundle
symfony/http-foundation       v4.2.1  v4.2.1  Symfony HttpFoundation Component
symfony/http-kernel           v4.2.1  v4.2.1  Symfony HttpKernel Component
symfony/inflector             v4.2.1  v4.2.1  Symfony Inflector Component
symfony/intl                  v4.2.1  v4.2.1  A PHP replacement layer for the C intl extension that includes additional data from the ICU library.
symfony/maker-bundle          v1.10.0 v1.10.0 Symfony Maker helps you create empty commands, controllers, form classes, tests and more so you can forget about writing boilerplate code.
symfony/monolog-bridge        v4.2.1  v4.2.1  Symfony Monolog Bridge
symfony/monolog-bundle        v3.3.1  v3.3.1  Symfony MonologBundle
symfony/options-resolver      v4.2.1  v4.2.1  Symfony OptionsResolver Component
symfony/orm-pack              v1.0.5  v1.0.5  A pack for the Doctrine ORM
symfony/panther               v0.2.0  v0.2.0  A browser testing and web scraping library for PHP and Symfony.
symfony/phpunit-bridge        v4.2.1  v4.2.1  Symfony PHPUnit Bridge
symfony/polyfill-intl-icu     v1.10.0 v1.10.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring     v1.10.0 v1.10.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php72        v1.10.0 v1.10.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/process               v4.2.1  v4.2.1  Symfony Process Component
symfony/profiler-pack         v1.0.4  v1.0.4  A pack for the Symfony web profiler
symfony/property-access       v4.2.1  v4.2.1  Symfony PropertyAccess Component
symfony/property-info         v4.2.1  v4.2.1  Symfony Property Info Component
symfony/routing               v4.2.1  v4.2.1  Symfony Routing Component
symfony/security-acl          v3.0.1  v3.0.1  Symfony Security Component - ACL (Access Control List)
symfony/security-bundle       v4.2.1  v4.2.1  Symfony SecurityBundle
symfony/security-core         v4.2.1  v4.2.1  Symfony Security Component - Core Library
symfony/security-csrf         v4.2.1  v4.2.1  Symfony Security Component - CSRF Library
symfony/security-guard        v4.2.1  v4.2.1  Symfony Security Component - Guard
symfony/security-http         v4.2.1  v4.2.1  Symfony Security Component - HTTP Integration
symfony/serializer            v4.2.1  v4.2.1  Symfony Serializer Component
symfony/serializer-pack       v1.0.2  v1.0.2  A pack for the Symfony serializer
symfony/stopwatch             v4.2.1  v4.2.1  Symfony Stopwatch Component
symfony/swiftmailer-bundle    v3.2.4  v3.2.4  Symfony SwiftmailerBundle
symfony/templating            v4.2.1  v4.2.1  Symfony Templating Component
symfony/test-pack             v1.0.5  v1.0.5  A pack for functional and end-to-end testing within a Symfony app
symfony/translation           v4.2.1  v4.2.1  Symfony Translation Component
symfony/twig-bridge           v4.2.1  v4.2.1  Symfony Twig Bridge
symfony/twig-bundle           v4.2.1  v4.2.1  Symfony TwigBundle
symfony/validator             v4.2.1  v4.2.1  Symfony Validator Component
symfony/var-dumper            v4.2.1  v4.2.1  Symfony mechanism for exploring and dumping PHP variables
symfony/var-exporter          v4.2.1  v4.2.1  A blend of var_export() + serialize() to turn any serializable data structure to plain PHP code
symfony/web-link              v4.2.1  v4.2.1  Symfony WebLink Component
symfony/web-profiler-bundle   v4.2.1  v4.2.1  Symfony WebProfilerBundle
symfony/webpack-encore-bundle v1.0.0  v1.0.0  Integration with your Symfony app & Webpack Encore!
symfony/yaml                  v4.2.1  v4.2.1  Symfony Yaml Component

```

#### PHP version

```
$ php -v
PHP 7.2.10-0ubuntu1 (cli) (built: Sep 13 2018 13:38:55) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.10-0ubuntu1, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.0, Copyright (c) 2002-2018, by Derick Rethans

```

## Steps to reproduce

- Updating sonata-project/core-bundle (3.12.0 => 3.13.2)

## Actual results

```
Argument 3 passed to Sonata\Twig\Extension\TemplateExtension::__construct() must be an instance of Sonata\CoreBundle\Model\Adapter\AdapterInterface, instance of Sonata\Doctrine\Adapter\AdapterChain given
```

**Comment by OskarStark** [comment]
Does this fix your issue?
https://github.com/sonata-project/sonata-doctrine-extensions/pull/107

**Comment by emr** [comment]
I have same issue. But using core-bundle with version 3.11, fixing the problem.

**Comment by axzx** [comment]
it looks like it has improved, but it's a different mistake:

Too few arguments to function Sonata\Form\Type\BasePickerType::__construct(), 0 passed in vendor/symfony/form/FormRegistry.php on line 92 and at least 1 expected

**Comment by OskarStark** [comment]
argh.. its a nightmare 👎

**Comment by pribeirojtm** [comment]
Here we go again. Reverting to 3.12.0 till stability in these kind of issues.

**Comment by gnutix** [comment]
Same issue here.

**Comment by danaki** [comment]
Lol, man, I'm watching this epic story since 3.13.0. New fix introduces a new bug.

**Comment by axzx** [comment]
@danaki I wonder how these changes passed the tests positively.

**Comment by emr** [comment]
@axzx Because "Deney noksanlığı" in Turkish. (Lack of experimentation/test)

**Comment by kunicmarko20** [comment]
Because we only have unit tests.

https://media1.tenor.com/images/6dc3aba8172530dda2407198466a3f2e/tenor.gif

![tenor](https://user-images.githubusercontent.com/13528674/50005366-0cd6f900-ffa2-11e8-8665-54183aa15984.gif)

**Comment by deantomasevic** [comment]
> Does this fix your issue?
> [sonata-project/sonata-doctrine-extensions#107](https://github.com/sonata-project/sonata-doctrine-extensions/pull/107)

Yep.

> composer install
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Nothing to install or update
Generating autoload files
Executing script cache:clear [OK]
Executing script assets:install public [OK]

Without [sonata-project/sonata-doctrine-extensions#107](https://github.com/sonata-project/sonata-doctrine-extensions/pull/107):
> !!  In TemplateExtension.php line 40:
!!
!!    Type error: Argument 3 passed to Sonata\Twig\Extension\TemplateExtension::_
!!    _construct() must be an instance of Sonata\CoreBundle\Model\Adapter\Adapter
!!    Interface, instance of Sonata\Doctrine\Adapter\AdapterChain given, called in
!!    /var/cache/dev/ContainerTicx8il/srcDevDebugProjectContainer.php on line 4289
!!

✌️ Dean

**Comment by 7system7** [comment]
Oh my god.. what's goin' on w/ this project??

**Comment by kunicmarko20** [comment]
Unsuccessfully try to split a common repository

**Comment by 7system7** [comment]
Oh yeah, the free-project-card! You won. :D Maybe it seems but I do not have. Btw, you really think, the world is limited to the Github? ;)

I have chosen this project because I assumed that the code is stable on every release. The latest releases  - finely phrased - are not. I thought that, the contributors have more prefessionality, than this, more than me. And after a couple of hour of debugging, it was a big disappointment to discover that the vendor code has a bug... (again)  

At the same time, everybody do mistakes. So, no offense. :) And I support for the success of the project!

![new-fingers-crossed-gif-466](https://user-images.githubusercontent.com/3964568/50044825-c52b9c80-0089-11e9-9d2f-fe1086fb9b7f.gif)

**Comment by greg0ire** [comment]
>  Btw, you really think, the world is limited to the Github? ;)

No, that's why I made sure to add "This might be a wrong deduction"

> I thought that, the contributors have more prefessionality, than this, more than me. 

I think I understand what you mean, but in case you don't know, we are professionals, but we don't work on this professionally, meaning we do this mostly evenings and weekends. For instance, I just worked a couple hours on this.

> At the same time, everybody do mistakes. So, no offense. :) And I support for the success of the project!

That's a far better attitude, thanks! I understand your disappointment, and we will think about ways to prevent this from happening again. Just keep in mind that indeed, shit happens and we do make mistakes, sometimes. A fix is available here: https://github.com/sonata-project/sonata-doctrine-extensions/pull/107 If you want to help, consider testing the solution I described in [this comment](https://github.com/sonata-project/sonata-doctrine-extensions/pull/107/files#r241951988)
